kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/arbitrary/OutputHexForUnprintableCharsTest.kt

```diff
@@ -0,0 +1,45 @@
+package com.sksamuel.kotest.property.arbitrary
+
+import io.kotest.assertions.AssertionFailedError
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.common.ExperimentalKotest
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+import io.kotest.property.Arb
+import io.kotest.property.PropTestConfig
+import io.kotest.property.arbitrary.Codepoint
+import io.kotest.property.arbitrary.ascii
+import io.kotest.property.arbitrary.printableAscii
+import io.kotest.property.arbitrary.string
+import io.kotest.property.checkAll
+import io.kotest.property.internal.escapeUnprintable
+
+@OptIn(ExperimentalKotest::class)
+class OutputHexForUnprintableCharsTest : StringSpec({
+
+   "should handle unprintable characters in failure messages" {
+      checkAll(
+         PropTestConfig(outputHexForUnprintableChars = true, seed = -5556043119863981334),
+         Arb.string(1..127, codepoints = Codepoint.ascii())
+      ) { str ->
+         val message = shouldThrow<AssertionFailedError> { str shouldBe "expectedString" }.message
+
+         val escapedMessage = message?.escapeUnprintable()
```

Comment:
presumably for some iterations if this test `message != escapedMessage`, correct? do we need to validate that assumption?

Reply:
I'm not sure if I understand what you mean, but let's take a look at my last commit and you can share your thoughts.

Reply:
> I'm not sure if I understand what you mean, but let's take a look at my last commit and you can share your thoughts.

that';s exactly what I meant, TY